packages/api-client/src/connections/mocks/MockCentralApi.ts

```diff
@@ -25,11 +72,18 @@ export class MockCentralApi implements CentralApiInterface {
   public createSurveyResponses(responses: MeditrakSurveyResponseRequest[]): Promise<void> {
     throw new Error('Method not implemented.');
   }
-  public fetchResources(
-    endpoint: string,
-    params?: Record<string, unknown> | undefined,
-  ): Promise<any> {
-    throw new Error('Method not implemented.');
+  public fetchResources(endpoint: string, params?: Params): Promise<any> {
+    return Promise.resolve(
+      this.mockData[endpoint]?.filter(resource => {
+        if (params?.filter) {
+          return Object.entries(params.filter)?.every(([field, filter]) => {
+            const resourceValue = resource[field];
+            return getValueMatchesFilter(resourceValue, filter);
+          });
+        }
+        return true;
+      }) ?? [],
+    );
```

Comment:
```suggestion
public async fetchResources(endpoint: string, params?: Params): Promise<any> {
    const resourceData = this.mockData[endpoint];
    if (!resourceData) return [];

    const filter = params?.filter;
    if (!filter) return resourceData;

    return resourceData.filter(resource =>
      Object.entries(filter).every(([field, check]) =>
        getValueMatchesFilter(resource[field], check),
      ),
    );
  }
```

Bit of personal preference on this one but I took a stab at rewriting this to be a bit more succinct. I'm a bit caution with using the optional chaining operator, as I find it can be hard to parse. So I prefer to exit early where it's convenient.